packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -1105,15 +795,23 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
         `SELECT ${pks} FROM ${table}`,
         options.where ? this.whereQuery(options.where, whereOptions) : '',
         `ORDER BY ${pks}`,
-        this._addLimitAndOffset(options),
+        this.#internals.addLimitAndOffset(options),
         ')',
       ]);
     }
 
     return joinSQLFragments([
       `DELETE FROM ${this.quoteTable(tableName)}`,
       options.where ? this.whereQuery(options.where, whereOptions) : '',
-      this._addLimitAndOffset(options),
+      this.#internals.addLimitAndOffset(options),
     ]);
   }
+
+  __TEST__getInternals() {
+    if (process.env.npm_lifecycle_event !== 'mocha') {
```

Comment:
I never knew this was possible. This is nice

Reply:
I'd like to go even further and figure out a way to strip the code from the production build using some sort of compiler comment, but no idea if that's even possible

I also considered using `process.env.NODE_ENV === 'test'`, but we don't use it, so checking whether we're running in mocha seems like a good solution as it's only meant for us